lib/ecco/rows_converter.rb

```diff
@@ -0,0 +1,50 @@
+module Ecco
+  module RowsConverter
+    def self.convert(java_rows, type)
+      if type == "UPDATE_ROWS"
+        convert_update_rows(java_rows)
+      else
+        convert_write_or_delete_rows(java_rows)
+      end
+    end
+
+    def self.convert_write_or_delete_rows(java_rows)
+      java_rows.to_a
+    end
+
+    def self.convert_update_rows(java_rows)
+      before_update_row = java_rows.first.get_key.to_a
+      after_update_row  = java_rows.first.get_value.to_a
+
+      Array.new(1) do
+        Tuple.new(
+          key: before_update_row,
+          value: after_update_row
+        )
+      end
+    end
+  end
+
+  class Tuple
```

Comment:
Maby we can come up with a better name for this class. What it contains are the column values before and after.

Can't say I have any good ideas right now though 🤔

Reply:
Just some ideas regarding this class. The name is still pending, but based on the findings in https://github.com/twingly/ecco/pull/57#discussion_r529630356 we should be able to come up with something descriptive.

```diff
diff --git a/lib/ecco/rows_converter.rb b/lib/ecco/rows_converter.rb
index 3b4582e..1f5b611 100644
--- a/lib/ecco/rows_converter.rb
+++ b/lib/ecco/rows_converter.rb
@@ -13,38 +13,29 @@ module Ecco
     end

     def self.convert_update_rows(java_rows)
-      before_update_row = java_rows.first.get_key.to_a
-      after_update_row  = java_rows.first.get_value.to_a
+      java_rows.map do |java_row|
+        before_update_row = java_row.get_key.to_a
+        after_update_row  = java_row.get_value.to_a

-      Array.new(1) do
-        Tuple.new(
-          key: before_update_row,
-          value: after_update_row
+        UpdateRowsEventUpdatedRow.new(
+          before: before_update_row,
+          after:  after_update_row
         )
       end
     end
   end

-  class Tuple
-    KEY   = 0
-    VALUE = 1
+  class UpdateRowsEventUpdatedRow
+    attr_reader :before
+    attr_reader :after

-    attr_reader :tuple
-
-    def initialize(key:, value:)
-      @tuple = [key, value]
-    end
-
-    def key
-      @tuple[KEY]
+    def initialize(before:, after:)
+      @before = before
+      @after  = after
     end

-    def value
-      @tuple[VALUE]
-    end
-
-    def inspect
-      @tuple.to_s
-    end
+    # Make sure we don't break existing code
+    alias_method :key, :before
+    alias_method :value, :after
   end
 end
```

Reply:
Note: I removed the inspect above, since I thought the default was good enough, but we could define our own if needed. What the default looks like:

```
#<Ecco::UpdateEventColumnsDiff:0x3667faa8 @after=[1, "another value"], @before=[1, "a value"]>
```

Reply:
I really like the idea of using aliases to make our code more readable from the inside and still work with the code from the binlog java stuff. Wish I had thought of that! :)